api/capi/include/nntrainer.h

```diff
@@ -358,6 +358,56 @@ int ml_train_dataset_create_with_generator(ml_train_dataset_h *dataset,
                                            ml_train_datagen_cb valid_cb,
                                            ml_train_datagen_cb test_cb);
 
+/**
+ * @brief Constructs the dataset.
+ * @details Use this function to create a dataset
+ * @since_tizen 6.5
+ * @remarks If the function succeeds, @a dataset must be released using
+ * ml_train_dataset_destroy(), if not added to a model. If added to a model, @a
+ * dataset is available until the model is released.
+ * @param[out] dataset The NNTrainer dataset handle.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid parameter.
+ */
+int ml_train_dataset_create(ml_train_dataset_h *dataset);
+
+/**
+ * @brief Add data generator callback to @a dataset
+ * @details Use this function to add a data generator callback which generates a
+ * single batch per call to the dataset.
+ * @param[in] dataset The NNTrainer dataset handle.
+ * @param[in] usage The phase where this generator should be used.
+ * @param[in] user_data user_data to be fed when @a cb is being called.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid parameter.
+ */
+int ml_train_dataset_add_generator(ml_train_dataset_h dataset,
+                                   ml_train_dataset_data_usage_e usage,
+                                   ml_train_datagen_cb cb, void *user_data);
+
+/**
+ * @brief Add data file to @a dataset
+ * @details Use this function to add a data file from where data is retrieved.
+ * @remarks %http://tizen.org/privilege/mediastorage is needed if @a dataset is
+ * saved to media storage.
+ * @remarks %http://tizen.org/privilege/externalstorage is needed if @a model is
+ * saved to external storage.
+ * @param[in] dataset The NNTrainer dataset handle.
+ * @param[in] usage The phase where this file should be used.
+ * @param[in] file file path.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid parameter.
```

Comment:
if the dataset already exist, the behavior is to overwrite? if yes, add as a comment here as well as in cb.

Reply:
dataset_usage train exist, it will be error. I'll make a comment here.

Reply:
so how does one change dataset set to a model. Does model allow setting a new dataset (replacing exist dataset)?

Reply:
setting dataset to a model is described in `ml_train_model_set_dataset()`. 

```
 * @remarks Unsets the previously set dataset, if any. The previously set
 * dataset must be freed using ml_train_dataset_destroy().
```

Reply:
cool then.